torch/quantization/fake_quantize.py

```diff
@@ -13,22 +13,20 @@ class FakeQuantize(Module):
         given the stats
     '''
 
-    def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
-                 quant_min=0, quant_max=255):
+    def __init__(self, observer=default_observer, quant_min=0, quant_max=255):
```

Comment:
with passing observer here it's getting a bit awkward because of nesting. Is FakeQuantize going to be the only base module? Can quant_min/quant_max be deduced from observer's dtype? If so - I'd recommend making FakeQuantize an always on wrapper (handled within prepare_qat) and having qconfig just point to observers instances

Reply:
We want to be able to experiment easily with different choices for quant min/max, say if we want to experiment with lower precisions for weights. In this case, observer may not have all information. Also, fake-quant can potentially learn scale and zero-point via backprop, i.e not have an observer within it. Ideally, we should structure fake-quant to have an abstract base class (like observers). This can be done after we land the basic implementations first.

Reply:
We can do:
```
    def __init__(self, observer=MinMaxObserver, quant_min=0, quant_max=255, **observer_kwargs):
        self.observer = observer(**observer_kwargs)

    QConfig(weight=FakeQuantize.with_args(dtype=np.quint8, quant_min=5))
```

Reply:
Done